platform-sdk/docs/proposals/001-merkle-state-api/001-merkle-state-api.md

```diff
@@ -0,0 +1,300 @@
+# Merkle State API
+
+## Summary
+
+Refactor the Merkle tree so that it can be fully represented by the State API objects.
+Create a module that will provide a simple and easy-to-use API for the Merkle tree.
+
+## Purpose and Context
+
+### Why?
+
+The Merkle tree is a rather complex data structure, and to make interaction easier, we need to hide this complexity behind the API.
+Another reason to have this API is to provide additional flexibility in changing the implementation details of the Merkle tree.
+As long as they are hidden, we are free to change them.
+
+Note that this API is already implemented as a part of the Service Modularization project, but these classes will be moved
+to a separate module as a part of this proposal.
+
+### Context
+
+There are two issues that need to be addressed:
+
+* We need to have a public API for the Merkle tree that is simple and easy to use, residing in a separate module with a minimal set of dependencies.
+* The Merkle tree has to be fully expressed in terms of State API. Currently, `PlatformState` stands out and needs to be refactored.
+
+#### Singleton
+
+A singleton is a node with 2 children:
+
+- string label for State Name
+- protobuf message for child data
+
+In Java code it's represented by the following classes:
+```java
+public interface ReadableSingletonState<T> {
+  @NonNull
+  String getStateKey();
+  T get();
```

Comment:
can you add a newline bethween the methods

Reply:
Done